include/swift/Basic/TaskQueue.h

```diff
@@ -21,6 +22,7 @@
 #include <functional>
 #include <memory>
 #include <queue>
+#include <sys/resource.h>
```

Comment:
This should not be in `include/swift/Basic/TaskQueue.h`; that file covers all platform task queues, including those non-unix / without-rusage. Instead, it should be in the `lib/Basic/Unix/TaskQueue.inc` unix-specific include fragment, and the code that includes this should be guarded by `#ifdef HAVE_SYS_RESOURCE_H`, and the code that _instantiates_ struct rusage should be guarded by `#if defined(HAVE_GETRUSAGE) && !defined(__HAIKU__)`

See the rusage-extracting code in `lib/Basic/Statistic.cpp` for example (also for the apple-specific bytes-vs-kb bit).

Reply:
It seems like the import isn't needed at all. It's at least building _(incremental and clean)_ without it.